opensquirrel/mckay_decomposer.py

```diff
@@ -3,135 +3,115 @@
+        self.decompose_and_add(q, p["angle"], p["axis"])
 
-    def _flush_all(self):
-        while len(self.oneQubitGates) > 0:
-            self._flush(next(iter(self.oneQubitGates)))
+    def flush_all(self):
+        while len(self.accumulated_1q_gates) > 0:
+            self.flush(next(iter(self.accumulated_1q_gates)))
 
-    def _acc(self, qubit, semantic: SingleQubitAxisAngleSemantic):
-        axis, angle, phase = semantic.axis, semantic.angle, semantic.phase
+    def accumulate(self, qubit, bloch_sphere_rotation: BlochSphereRotation):
+        axis, angle, phase = bloch_sphere_rotation.axis, bloch_sphere_rotation.angle, bloch_sphere_rotation.phase
 
-        if qubit not in self.oneQubitGates:
-            self.oneQubitGates[qubit] = {"angle": angle, "axis": axis, "phase": phase}
+        if qubit not in self.accumulated_1q_gates:
+            self.accumulated_1q_gates[qubit] = {"angle": angle, "axis": axis, "phase": phase}
             return
 
-        existing = self.oneQubitGates[qubit]
-        combinedPhase = phase + existing["phase"]
+        existing = self.accumulated_1q_gates[qubit]
+        combined_phase = phase + existing["phase"]
 
         a = angle
         l = axis
         b = existing["angle"]
         m = existing["axis"]
 
-        combinedAngle = 2 * acos(cos(a / 2) * cos(b / 2) - sin(a / 2) * sin(b / 2) * np.dot(l, m))
+        combined_angle = 2 * acos(cos(a / 2) * cos(b / 2) - sin(a / 2) * sin(b / 2) * np.dot(l, m))
 
-        if abs(sin(combinedAngle / 2)) < ATOL:
-            self.oneQubitGates.pop(qubit)
+        if abs(sin(combined_angle / 2)) < ATOL:
+            self.accumulated_1q_gates.pop(qubit)
             return
 
-        combinedAxis = (
+        combined_axis = (
             1
-            / sin(combinedAngle / 2)
+            / sin(combined_angle / 2)
             * (sin(a / 2) * cos(b / 2) * l + cos(a / 2) * sin(b / 2) * m + sin(a / 2) * sin(b / 2) * np.cross(l, m))
         )
 
-        self.oneQubitGates[qubit] = {"angle": combinedAngle, "axis": combinedAxis, "phase": combinedPhase}
+        self.accumulated_1q_gates[qubit] = {"angle": combined_angle, "axis": combined_axis, "phase": combined_phase}
 
-    def process(self, squirrelAST):
-        # FIXME: duplicate gates in ast and self??
-        self.output = SquirrelAST(self.gates, squirrelAST.nQubits, squirrelAST.qubitRegisterName)
-        self.oneQubitGates = {}
+    def process_gate(self, gate: Gate):
+        qubit_arguments = [arg for arg in gate.arguments if isinstance(arg, Qubit)]
 
-        for operation in squirrelAST.operations:
-            if isinstance(operation, str):
-                continue
-
-            self._processSingleOperation(operation)
+        if len(qubit_arguments) >= 2:
+            [self.flush(q) for q in qubit_arguments]
+            self.output.add_gate(gate)
+            return
 
-        self._flush_all()
+        if len(qubit_arguments) == 0:
+            assert False, "Unsupported"
+            return
```

Comment:
PyCharm complains that the `return` statement is unreachable, does the code exit when you have `assert False`?